ethpm_types/contract_type.py

```diff
@@ -204,6 +205,51 @@ def extract_sourcemap_item(expanded_row, item_idx, previous_val=None):
             yield item
 
 
+class PCMapItem(BaseModel):
+    line_start: Optional[int] = None
+    column_start: Optional[int] = None
+    line_end: Optional[int] = None
+    column_end: Optional[int] = None
+
+
+class PCMap(BaseModel):
+    """
+    As part of the source output, the compiler provides a map of program counter values to
+    statements in the source code that the instructions were compiled from.
+
+    This can be used for various purposes ranging from static analysis tools that
+    report errors based on the program counter value and debugging tools that highlight local
+    variables and their uses.
+    """
+
+    __root__: Dict[str, List[int]]
+
+    def parse(self) -> Dict[int, PCMapItem]:
+        """
+        Parses the pc map string into a map of ``PCMapItem`` items, using integer pc values as keys.
+
+        The format from the compiler will have numeric string keys with lists of ints for values.
```

Comment:
Every time I see a dict where `int` is the key, I ask myself if this could be a list instead.
Could this be a list instead? Where PC values are the indices?

Reply:
I guess it's not really `ints` but PCMapItems (keys with int values)...

Reply:
I guess I can probably answer my own question... There is probably gaps in PC lines so it makes more sense as a dict

Reply:
> There are probably gaps in PC lines so it makes more sense as a dict

Yup exactly. Also this is why I went with dict because we're getting arbitrary PC values from the trace (e.g. 186 or something) and need to then find the corresponding line information

With a dict we can then just index `pcmap[pc_num]` rather than having to iterate on anything. Originally I had `pc` values on the `PCMapItems` but yeah same thing basically, you'd have to loop through the list (ideally backwards though!) to find the relevant item

cc @NotPeopling2day